tests/elster_xml/grundsteuer/test_elster_data_representation.py

```diff
@@ -7,12 +7,256 @@
+        gebaeude = SampleGebaeude().with_wohnflaeche(100).parse()
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Wohn_Unter60 is None
+        assert result.Wohn_60bis100 is None
+        assert result.Wohn_ab100 is not None
+
+    def test_on_weitere_wohnraeume_flag_true_should_set_weitere_wohnraeume(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(10).with_weitere_wohnraeume(42, 2).parse()
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Weitere_Wohn is not None
+
+    def test_on_weitere_wohnraeume_flag_false_should_not_set_weitere_wohnraeume(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(10).with_weitere_wohnraeume(42, 2).parse()
+        gebaeude.weitere_wohnraeume.has_weitere_wohnraeume = False
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Weitere_Wohn is None
+
+
+class TestEAngWohn:
+    def test_on_ab_1949_should_contain_baujahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_baujahr("1959").parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403114 == "1959"
+
+    def test_on_vor_1949_should_not_contain_baujahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_baujahr("1959").parse()
+        gebaeude.ab1949.is_ab1949 = False
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403114 is None
+
+    def test_on_kenrsaniert_should_contain_kernsanierungsjahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_kernsanierung("1959").parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403115 == "1959"
+
+    def test_on_not_kenrsaniert_should_not_contain_kernsanierungsjahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_kernsanierung("1959").parse()
+        gebaeude.kernsaniert.is_kernsaniert = False
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403115 is None
+
+    def test_on_abbruchverpflichtung_should_contain_abbruchverpflichtungsjahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_abbruchverpflichtung("1959").parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403116 == "1959"
+
+    def test_on_no_abbruchverpflichtung_should_not_contain_abbruchverpflichtungsjahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_abbruchverpflichtung("1959").parse()
+        gebaeude.abbruchverpflichtung.has_abbruchverpflichtung = False
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403116 is None
+
+    def test_on_garagen_should_contain_set_anzahl(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_garagen(2).parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.Garagen.E7403171 == 2
+
+    def test_on_no_garagen_should_contain_not_contain_garagen(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_garagen(2).parse()
+        gebaeude.garagen.has_garagen = False
```

Comment:
Why do you first set garagen just to then unset them?

Reply:
To make it explicit and visible what's going on here. We construct a valid input object with the garage flag and anzahl set, then explicitly set the flag to false to create the invalid state we'd like to test.

Reply:
Mh..although this would be an invalid input to begin with? I feel like we don't need to and even shouldn't test that. I'd like to rely on the input being valid (with regards to the input validators only, of course).

What might be interesting: What if I input "garagen: true, anzahl: 0"? Do we catch that anywhere? Mh...but even if not, we'll get an error from Eric, so it should be fine. But we should think about that in the webapp.